### ISSUE TITLE
Implement in-place MPI reductions

### ISSUE THREAD

**Issue by galexv** [ALPSCore/ALPSCore#386]
Required, e.g., by ALEA: https://github.com/ALPSCore/ALPSCore/blob/0b04f2508ddbaaebe6bf878b48aa4b3fb1b0692d/alea/include/alps/alea/mpi.hpp#L81

**Comment by mwallerb** [comment]
Yeah, so ALEA works with the underlying MPI library, but of course it would be nice to have it in the wrapper . . .  I would say it is rather low prio?
